src/app/core/components/header/header.component.ts

```diff
@@ -8,26 +11,44 @@ import { UserEntity } from '../../models/user-entity.model';
 })
 export class HeaderComponent implements OnInit {
 
-  @Output() open = new EventEmitter<void>();
-  @Input() isAuth = true;
+  isAuth = true;
+  aSub: Subscription;
+  user : UserEntity;
+  // user : UserEntity = {
+  //   login: '1',
+  //   id: 9999,
+  //   password: 'fdfds',
+  //   first: 'fdsfsdf',
+  //   last: 'fddfdf'
+  // }
+  currentId: number;
 
   public content: string = 'content here';
-  public usersEntity: Array<UserEntity> = [
-    {
-      id: 1,
-      firstName: 'Alex',
-      lastName: 'Werby',
-    },
-    {
-      id: 2,
-      firstName: 'Aliaksandr',
-      lastName: 'Jastine',
-    }
-  ]
 
-  constructor() {}
+  constructor(
+    private auth: AuthorizationService,
+    private router: Router,
+  ) { }
 
   ngOnInit() {
+    this.router.events.subscribe((event) => {
+      if (event instanceof NavigationEnd) {
+        this.updateId();
+      }
+    });
+  }
+
+  updateId(): void {
+    this.currentId = this.auth.getCurrentId();
+    if (this.currentId !== undefined) {
```

Comment:
I prefer checking if(!this.currentId) // check on null , undefined, empty string. cause all such cases if occur, won't work further.

Reply:
changed code